src/utils/historyUtils.ts

```diff
@@ -58,30 +58,35 @@ export function formatHistoryDate (date: Date): string {
   return moment(date).format('llll')
 }
 
-export interface OldHistoryEntry {
-    id: string;
-    text: string;
-    time: number;
-    tags: string[];
-    pinned: boolean;
+export interface V0HistoryEntry {
+  id: string;
+  text: string;
+  time: number;
+  tags: string[];
+  pinned: boolean;
+}
+
+export function convertV0History (oldHistory: V0HistoryEntry[]): HistoryEntry[] {
+  return oldHistory.map((entry: V0HistoryEntry) => {
+    return {
+      id: entry.id,
+      title: entry.text,
+      lastVisited: moment(entry.time).toDate(),
+      tags: entry.tags,
+      pinned: entry.pinned
+    }
+  })
 }
 
 export function loadHistoryFromLocalStore (): HistoryEntry[] {
   const historyJsonString = window.localStorage.getItem('history')
+
   if (!historyJsonString) {
     // if localStorage["history"] is empty we check the old localStorage["notehistory"]
     // and convert it to the new format
     const oldHistoryJsonString = window.localStorage.getItem('notehistory')
-    const oldHistory = oldHistoryJsonString ? JSON.parse(JSON.parse(oldHistoryJsonString)) as OldHistoryEntry[] : []
-    return oldHistory.map((entry: OldHistoryEntry) => {
-      return {
-        id: entry.id,
-        title: entry.text,
-        lastVisited: moment(entry.time).toDate(),
-        tags: entry.tags,
-        pinned: entry.pinned
-      }
-    })
+    const oldHistory = oldHistoryJsonString ? JSON.parse(JSON.parse(oldHistoryJsonString)) as V0HistoryEntry[] : []
```

Comment:
Wait, we're parsing the JSON twice here? Why?

Reply:
![image](https://user-images.githubusercontent.com/6124140/83802154-8ee14300-a6aa-11ea-90e7-163de182250f.png)

Reply:
Because codimd stringifies the history array twice.

Reply:
Argh, okay.

Reply:
yeah, we do that because the old frontend `JSON.stringfy`ed twice